guevaraTorres/assignments/cyclicSort/lc287/lc287.js

```diff
@@ -0,0 +1,87 @@
+// Problem Statement #
+
+// We are given an unsorted array containing ‘n+1’ numbers taken from the range 1 to ‘n’. The array has only one duplicate but it can be repeated multiple times. Find that duplicate number without using any extra space. You are, however, allowed to modify the input array.
+
+
+/**
+ * 
+ * @param {*} nums 
+ * @returns
+ * 
+ * Initialize a pointer
+ * Then loop through the array
+ * Checking for a duplicate of one number
+ * 
+ * currPtr = 0, nums(currPtr) = 1;
+ * currPtr = 1, nums(currPtr) = 4;
+ * if(value of currPtr !== currPtr + 1) currPtr++;
+ * if it does, then return currPtr
+ * 
+ * edge cases: array with no elements or duplicates, return empty array 
+ */
+
+const find_duplicate = function(nums) {
+  // TODO: Write your code here
+
+  let currPtr = 0;                                          //current pointer is initialized
+
+  if(nums.length < 2) return 0
+  while(currPtr < nums.length) {                            //while pointer is less than length of numbers array
+    if(nums[currPtr] !== currPtr + 1){                      //if current index does not equal next index
```

Comment:
Instead of `//if current index does not equal next index`, it is more `//if current value at our current index does not equal next index`